sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java

```diff
@@ -83,12 +83,24 @@ public SpanBuilder setSpanKind(SpanKind spanKind) {
 
   @Override
   public SpanBuilder addLink(SpanContext spanContext) {
+    if (spanContext == null || !spanContext.isValid()) {
+      return this;
+    }
     addLink(LinkData.create(spanContext));
     return this;
   }
 
   @Override
   public SpanBuilder addLink(SpanContext spanContext, Attributes attributes) {
+    if (spanContext == null || !spanContext.isValid()) {
+      return this;
+    }
+    if (!spanContext.isValid()) {
```

Comment:
this is redundant with the check above, correct?

Reply:
Oops